lib/workflow/messengers/messenger-AMQP.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = amqpMessengerFactory;
+amqpMessengerFactory.$provide = 'Task.Messengers.AMQP';
+amqpMessengerFactory.$inject = [
+    'Protocol.Task',
+    'Protocol.Events',
+    'Protocol.TaskGraphRunner',
+    'Promise'
+];
+function amqpMessengerFactory(
+    taskProtocol,
+    eventsProtocol,
+    taskGraphRunnerProtocol,
+    Promise
+) {
+    var exports = {};
+
+    exports.subscribeRunTask = function(domain, callback) {
+        return taskProtocol.subscribeRun(domain, callback);
+    };
+
+    exports.publishRunTask = function(domain, taskId, graphId) {
+        return taskProtocol.run(domain, { taskId: taskId, graphId: graphId });
+    };
+
+    exports.subscribeCancel = function(callback) {
+        return taskProtocol.subscribeCancel(callback);
+    };
+
+    exports.publishCancelTask = function(taskId, errName, errMessage) {
+        return taskProtocol.cancel(taskId, errName, errMessage);
+    };
+
+    exports.subscribeTaskFinished = function(domain, callback) {
+        return eventsProtocol.subscribeTaskFinished(domain, callback);
+    };
+
+    exports.publishTaskFinished = function(
+            domain,
+            taskId,
+            graphId,
+            state,
+            context,
+            terminalOnStates
+        ) {
+        return eventsProtocol.publishTaskFinished(
+                domain,
+                taskId,
+                graphId,
+                state,
+                context,
+                terminalOnStates
+        );
+    };
+
+    exports.subscribeRunTaskGraph = function(domain, callback) {
```

Comment:
Yeah, we don't leverage this event internally but someone could publish it externally via their own AMQP driver.